File: backend/src/main/java/org/example/backend/service/BlackListService.java
package org.example.backend.service;

import org.example.backend.model.entity.User;

import java.util.List;

public interface BlackListService {
    void blockUser(Long userId, Long chatId);

    void unblockUser(Long userId, Long chatId);

    Boolean isUserBlocked(Long chatId);

    List<User> getBlackList(Long chatId);
}

File: frontend/src/app/store/blacklist-store/service/blacklist.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {environment} from "../../../../environments/environment";
import {User} from "../../user-store/service/user.model";

@Injectable({
  providedIn: 'root'
})
export class BlacklistService {
  private apiUrl = environment.baseUrl + "/api/blacklist"

  constructor(private http: HttpClient) {
  }

  blockUser(userId: number | string, chatId: number | string) {
    return this.http.post(this.apiUrl + "/block", {userId, chatId})
  }

  unblockUser(userId: number | string, chatId: number | string) {
    return this.http.post(this.apiUrl + "/unblock", {userId, chatId})
  }

  getBlackList(chatId: number | string) {
    return this.http.get<User[]>(this.apiUrl + '/chat/' + chatId)
  }
}
